### PR TITLE
Configurabe default router publishing

### PR DESCRIPTION
**Pull request by janosi**
In some deployment situations we should be able to skip the automatic creation of a LoadBalancer Service in front of the default router. The OpenShift IngressController API has the necessary EndpointPublishingStartegy Type for this purpose: "Private".https://github.com/openshift/api/blob/16dd969d555004ba13853ccc2e677d098189ae61/operator/v1/types_ingress.go#L330  When "Private" is set the IngressController operator does not create any Service that would expose the default router externally.

This PR enables the configurability of the EndpointPublishingStartegy Type via the pkg/api/ClusterParams.RouterServiceType parameter https://github.com/openshift/ibm-roks-toolkit/blob/8ba3b24e6819b97eb83d04c74f7638fde2881be5/pkg/api/types.go#L29 If that parameter is set to the new "Private" value the IngressController.EndpointPublishingStartegy.Type gets the value "Private". Otherwise the old Ingresscontroller setup is applied.

This PR is the replacement of https://github.com/openshift/ibm-roks-toolkit/pull/246

### CODE REVIEW

File: assets/cluster-bootstrap/cluster-ingresscontrollers-02-config.yaml [hunk 1]
```diff
@@ -4,12 +4,15 @@ metadata:
   name: default
   namespace: openshift-ingress-operator
 spec:
-{{ if .EndpointPublishingStrategyScope }}
+{{- if eq .RouterServiceType "Private" }}
+  endpointPublishingStrategy:
+    type: Private 
+{{- else if .EndpointPublishingStrategyScope }}
```
Comment (rtheis) [inline assets/cluster-bootstrap/cluster-ingresscontrollers-02-config.yaml:10]: I think that I follow the plan here.  Basically `RouterServiceType` can be either `Private` or `LoadBalancer` and `EndpointPublishingStrategyScope` is used when `RouterServiceType` is `LoadBalancer`.  Given this, it seems best if we cleanup the interface here.  I propose changing `{{- else if .EndpointPublishingStrategyScope }}` to `{{- else if eq .RouterServiceType "LoadBalancer" }}`.  So if an LB router is requested, `EndpointPublishingStrategyScope` must be set.  What do you think?

Comment (janosi) [reply assets/cluster-bootstrap/cluster-ingresscontrollers-02-config.yaml:10]: I agree, that would be a much cleaner logic here. The reason for the current setup is that I wanted to make the least change necessary. In order not to break anything.
I believe you have a much better understanding on how the new setup you proposed would apply for example when existing clusters are upgraded. If you are OK with the setup you proposed, I am fine, too, and I can update it accordingly.

Comment (janosi) [reply assets/cluster-bootstrap/cluster-ingresscontrollers-02-config.yaml]: Updated the yaml. Let's see if it meets your expectations. I kept `EndpointPublishingStrategyScope` optional, as it is optional in the API definition, too.

Comment (rtheis) [reply assets/cluster-bootstrap/cluster-ingresscontrollers-02-config.yaml]: I like what you've done here.  It is cleaner and is backwards compatible with what we have today.

**Review by rtheis — APPROVED** [review]
/lgtm
/approved

**Pull request merged**